nixos/modules/services/web-apps/lemmy.nix

```diff
@@ -0,0 +1,142 @@
+{ lib, pkgs, config, ... }:
+with lib;
+let
+  cfg = config.services.lemmy;
+  settingsFormat = pkgs.formats.json { };
+in
+{
+  options.services.lemmy = {
+    enable = mkEnableOption "Enable lemmy server";
+
+    config = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options.database = {
+          password = mkOption {
```

Comment:
I'm not 100% but we should be able to use the unix socket trick
https://github.com/diesel-rs/diesel/blob/3f3ce57ca1a89274ecf915b70826234c45bd261e/diesel/src/mysql/connection/url.rs#L216
this is how the library handles mysql, but it's probably the same for postgres.
Just leaving this comment so we don't forget to try.

Reply:
From the doc here https://www.postgresql.org/docs/9.4/libpq-connect.html
it should probably be something like
`postgresql:///run/postgresql`